libs/java/server_common/src/main/java/com/yahoo/athenz/common/messaging/DomainChangeMessage.java

```diff
@@ -0,0 +1,156 @@
+/*
+ *
+ *  * Copyright The Athenz Authors
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *     http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package com.yahoo.athenz.common.messaging;
+
+import java.util.Objects;
+
+public class DomainChangeMessage {
+    
+    private String domainName;
+    
+     // Domain change message id     
+    private String uuid;
+   
+    // The number of milliseconds since the epoch 
+    private long published;
+    
+    // Change in domain meta
```

Comment:
@dvirguttman how about making this more generic? Instead of having each one as a separate field, why not just have a couple of attribute with enums with additional details. I'm thinking about:

objectType: { DOMAIN, ROLE, POLICY, SERVICE, ENTITY}
objectName: here we can be more specific -e.g. role or policy name based on the object Type
operation: {api name - putRole, putRoleMeta, putPolicy, etc} - this way I know what type of operation was carried out - .e.g if I have:

objectType: ROLE
objectName: admin
api: putRoleMeta

then I know that metadata for the admin role was modified. This should give us more details and flexibility to add details